rpc/v9/transaction.go

```diff
@@ -552,62 +540,43 @@ func (h *Handler) TransactionByBlockIDAndIndex(
 //
 // It follows the specification defined here:
 // https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L222
-func (h *Handler) TransactionReceiptByHash(hash felt.Felt) (*TransactionReceipt, *jsonrpc.Error) {
-	var (
-		preconfirmedB      *core.Block
-		preconfirmedBIndex int
-	)
+func (h *Handler) TransactionReceiptByHash(hash *felt.Felt) (*TransactionReceipt, *jsonrpc.Error) {
+	if pending, err := h.PendingData(); err == nil {
+		receipt, parentHash, blockNumber, err := pending.ReceiptByHash(hash)
+		if err == nil {
+			txn, err := pending.TransactionByHash(hash)
+			if err == nil {
+				status := TxnPreConfirmed
+				if parentHash != nil {
+					// preLatest block
+					status = TxnAcceptedOnL2
+				}
+				return AdaptReceipt(receipt, txn, status, nil, blockNumber), nil
+			}
+		}
+	}
```

Comment:
I'm wondering if we can move this logic into a separate function, because the condition is quite nested. Something like a `extractPendingReceipt` function and inside that, we return nil, err everytime err != nil.